### DISCUSSION TITLE
manipulate remaining stock to receive on a PO

### DISCUSSION THREAD

**Discussion by anonkwe** [inventree/InvenTree#6680] [Q&A]
Is there a way that I can manipulate the logic of stock reception to be kind of a one shot one deal? What I mean by that is if we are supposed to receive ,let’s say 10 units,  but we finally receive less than that, like 9 , is where can I tweak the code to make it fully received received 9/9 instead of 9/10, In order to close the line item as fully received even if its 9/9. Problem we encounter right now is that the real quantity we are receiving always differs from reality. 

long story short, in models.py from order, i would like this action to automatically close the line item to be received (line.received += quantity would become the total quantity no matter what), so it would automatically modify the initial quantity of the po as well and fully receive the lineitem:

![image](https://github.com/inventree/InvenTree/assets/61838766/fc149d40-711c-4fc0-8042-29b915c5192e)

![image](https://github.com/inventree/InvenTree/assets/61838766/fd572657-dd1c-4df4-9a8c-f112e405c7ab)

**Comment by SchrodingersGat** [comment]
### Plugin
This could be potentially implemented with a plugin without adjusting the existing behaviour. This would be the "simplest" approach as we do not have to consider how to retain current behaviour and UX

### Optional Field

An optional "update_total_quantity" field in the "receive items" serializer could potentially be implemented, allowing user to select dynamically if a particular line item total quantity should be updated
